src/Workspaces/Core/Desktop/Workspace/MSBuild/MSBuildProjectLoader.cs

```diff
@@ -276,17 +280,50 @@ public LoadState(IReadOnlyDictionary<string, ProjectId> projectPathToProjectIdMa
             public void Add(ProjectInfo info)
             {
                 _projectIdToProjectInfoMap.Add(info.Id, info);
-                _projectInfoList.Add(info);
+                //Memoized results of ProjectAlreadyReferencesProject may no longer be correct;
+                //reset the cache.
+                ProjectAlreadyReferencesProjectMemo = new Dictionary<ProjectId, Dictionary<ProjectId, bool>>();
```

Comment:
You could just call .Clear() instead.

---
In reply to: [81627928](https://github.com/dotnet/roslyn/pull/14245#discussion_r81627928) [](ancestors = 81627928)